### PR TITLE
clients: more meaningful client version reporting

### PR DESCRIPTION
**Pull request by renaynay**
This PR changes the way clients report their version to hive. They report to a txt file instead of a json file.

### CODE REVIEW

File: clients/nethermind/Dockerfile [hunk 1]
```diff
@@ -16,7 +16,7 @@ RUN chmod +x /nethermind.sh
 RUN chmod +x /enode.sh
 
 # Write the version file.
-RUN echo "{\"repo\":\"https://github.com/nethermindeth/nethermind\",\"branch\":\"\",\"commit\":\"$(cat /nethermind/git-hash)\"}" > /version.json
+RUN echo "repo: https://github.com/nethermindeth/nethermind, commit: "$(cat /nethermind/git-hash)"" > /version.txt
```
Comment (fjl) [inline clients/nethermind/Dockerfile:19]: We don't need to put the repo in here. Would be better to have some kind of version number instead.

File: clients/trinity/Dockerfile [hunk 2]
```diff
@@ -28,7 +28,7 @@ RUN \
   | jq ".+ {\"repo\":\"$(git config --get remote.origin.url)\"}" \
   | jq ".+ {\"branch\":\"$(git rev-parse --abbrev-ref HEAD)\"}"  \
   | jq ".+ {\"commit\":\"$(git rev-parse HEAD)\"}"               \
-  > /version.json
+  > /version.txt
```
Comment (fjl) [inline clients/trinity/Dockerfile:31]: Let's remove the JSON here.

File: clients/besu/Dockerfile [hunk 3]
```diff
@@ -17,11 +17,7 @@ ADD genesis.json /genesis.json
 ADD enode.sh /enode.sh
 RUN chmod +x /enode.sh
 
-RUN echo "{\
-  \"repo\":\"https://github.com/hyperledger/besu\",\
-  \"branch\":\"${branch:-detached}\", \
-  \"commit\":\"${VCS_REF:-xxxxxx}\"\
-}" > /version.json
+RUN echo "repo: https://github.com/hyperledger/besu, branch: "${branch:-detached}", commit:"${VCS_REF:-xxxxxx}"" > /version.txt
```
Comment (fjl) [inline clients/besu/Dockerfile:20]: We can remove the repo URL here, and try to add a version number instead.

File: clients/aleth/Dockerfile [hunk 4]
```diff
@@ -17,7 +17,7 @@ ADD enode.sh /enode.sh
 RUN chmod +x /enode.sh
 
 # Copy buildinfo into version.json, and also remove the 'bool' field is_prerelease.
-RUN jq "del(.is_prerelease)" /usr/share/aleth/buildinfo.json > /version.json
+RUN jq "del(.is_prerelease)" /usr/share/aleth/buildinfo.json > /version.txt
```
Comment (fjl) [inline clients/aleth/Dockerfile:20]: Can we transform this JSON file to human-readable text somehow?

File: clients/besu/Dockerfile (hunk 3 above)
Comment (renaynay) [reply clients/besu/Dockerfile]: Can I just shovel output from `<client> --version` > /version.txt ? @fjl

**Comment by fjl** [comment]
Here's what this outputs right now:

```
  "clientVersions": {
    "aleth": "aleth 1.8.0-5+commit.9ab7d09a.dirty\neth network protocol version: 63\nClient database version: 12041\nBuild: linux/release\n",
    "besu": "besu/v21.1.1-dev-a75e87ab/linux-x86_64/oracle_openjdk-java-11\n",
    "go-ethereum": "Geth\nVersion: 1.9.26-unstable\nGit Commit: 54735a67239d19d3f29e3f8ebd348a6c88ed31fa\nGit Commit Date: 20210203\nArchitecture: amd64\nGo Version: go1.15.7\nOperating System: linux\nGOPATH=\nGOROOT=go\n",
    "nethermind": "Nethermind/v1.10.17-11-c9899ba0c-20210205/X64-Linux/5.0.2\n",
    "openethereum": "OpenEthereum Client.\n  version OpenEthereum/v3.1.1-stable/x86_64-linux-musl/rustc1.44.0\nCopyright 2015-2020 Parity Technologies (UK) Ltd.\nLicense GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.\nThis is free software: you are free to change and redistribute it.\nThere is NO WARRANTY, to the extent permitted by law.\n\nBy Wood/Paronyan/Kotewicz/Drwięga/Volf/Greeff\n   Habermeier/Czaban/Gotchac/Redman/Nikolsky\n   Schoedon/Tang/Adolfsson/Silva/Palm/Hirsz et al.\n\n"
  },
```

I think we can improve that. 

- For aleth, we only need the first line.
- Besu is OK
- Geth --version output is too messy
- Nethermind is OK
- OpenEthereum has way too much useless info, we only need the second line with the "   version " prefix removed.

**Comment by fjl** [comment]
We should also run `strings.TrimSpace` on the version.txt content to remove the extra newline.

**Comment by fjl** [comment]
You can get this kind of version string from geth using

    geth console --exec 'console.log(admin.nodeInfo.name)' --maxpeers=0 --nodiscover 2>/dev/null | head -1

**Comment by renaynay** [comment]
Current cc @fjl 

```
  "clientVersions": {
    "aleth": "\"1.8.0-5+commit.9ab7d09a.dirty\"\n",
    "besu": "besu/v21.1.1-dev-a75e87ab/linux-x86_64/oracle_openjdk-java-11\n",
    "go-ethereum": " / Version: 1.9.26-unstable / Go Version: go1.15.8 / Architecture: amd64\n",
    "nethermind": "Nethermind/v1.10.17-11-c9899ba0c-20210205/X64-Linux/5.0.2\n",
    "openethereum": "OpenEthereum/v3.1.1-stable/x86_64-linux-musl/rustc1.44.0\n"
  },
```

File: clients/aleth/Dockerfile [hunk 5]
```diff
@@ -17,7 +17,7 @@ ADD enode.sh /enode.sh
 RUN chmod +x /enode.sh
 
 # Copy buildinfo into version.json, and also remove the 'bool' field is_prerelease.
-RUN jq "del(.is_prerelease)" /usr/share/aleth/buildinfo.json > /version.json
+RUN jq '.version' /usr/share/aleth/buildinfo.json > /version.txt
```
Comment (fjl) [inline clients/aleth/Dockerfile:20]: You can use `jq -r '.version'` here to remove the quotes

**Pull request merged**